### PR TITLE
fix 23034: added Variable for Navbar padding

### LINKED ISSUE #23034: Update responsive navbar padding to use variables
Currently .nabvar-expand-* classes adds padding to .nav-link wrapped by `media-breakpoint-up`
https://github.com/twbs/bootstrap/blob/v4-dev/scss/_navbar.scss#L168-L171

This should be moved few lines above to `media-breakpoint-down`  https://github.com/twbs/bootstrap/blob/v4-dev/scss/_navbar.scss#L143 

It would be great if those paddings were variables.

EDIT:
should be moved to `media-breakpoint-down`

### PR DESCRIPTION
**Pull request by chiraggmodi**
I have added `$nav-link-padding` in `variables.scss`. Fixes #23034.

### CODE REVIEW

File: scss/_list-group.scss [hunk 1]
```diff
@@ -85,6 +85,7 @@
 
 .list-group-flush {
   .list-group-item {
+    border-top: 0;
```
Comment (wojtask9) [inline scss/_list-group.scss]: this change is unrelated.

File: scss/_variables.scss [hunk 2]
```diff
@@ -528,6 +528,7 @@ $zindex-tooltip:            1070 !default;
 
 // Navs
 
+$nav-link-padding:              .5rem !default;
```
Comment (wojtask9) [inline scss/_variables.scss]: I would call this $navbar-link-padding and put with others navbar-* variables.
Also consider if split between x/y axis wouldn't be better

File: scss/_list-group.scss (hunk 1 above)
Comment (chiraggmodi) [inline scss/_list-group.scss]: yes sorry for that, actually this is for another PR

File: scss/_variables.scss (hunk 2 above)
Comment (chiraggmodi) [inline scss/_variables.scss]: yes that would be better will update this changes are PR again.

Comment (chiraggmodi) [inline scss/_variables.scss]: I have made your requested changes, but for navbar link variable i have splited  in left/right instead of x/y as this are for left and right side not left/right and top/bottom.

**Comment by chiraggmodi** [comment]
because if some folks wants to give different value then they can overwrite those variable.

**Comment by gijsbotje** [comment]
I get that but then all the values with `-x` and `-y` would have to be changed to
I would suggest to leave this to the programmer to overwrite this in custom scss and just stick with `-x`

**Comment by chiraggmodi** [comment]
have change `$navbar-link-padding` variable to `$navbar-link-padding-x`

**Comment by mdo** [comment]
This was superseded by #25117
